receiver/azureeventhubreceiver/eventhubhandler.go

```diff
@@ -71,60 +71,33 @@ func (h *eventhubHandler) run(ctx context.Context, host component.Host) error {
 			h.settings.Logger.Debug("Error connecting to Event Hub", zap.Error(newHubErr))
 			return newHubErr
 		}
-		h.hub = &hubWrapperImpl{
-			hub: hub,
-		}
+		h.hub = &hubWrapperImpl{hub: hub}
 	}
 
-	if h.config.Partition == "" {
-		// listen to each partition of the Event Hub
-		runtimeInfo, err := h.hub.GetRuntimeInformation(ctx)
-		if err != nil {
-			h.settings.Logger.Debug("Error getting Runtime Information", zap.Error(err))
-			return err
-		}
-
-		for _, partitionID := range runtimeInfo.PartitionIDs {
-			err = h.setUpOnePartition(ctx, partitionID, false)
-			if err != nil {
-				h.settings.Logger.Debug("Error setting up partition", zap.Error(err))
-				return err
-			}
-		}
-	} else {
+	if h.config.Partition != "" {
 		err := h.setUpOnePartition(ctx, h.config.Partition, true)
 		if err != nil {
 			h.settings.Logger.Debug("Error setting up partition", zap.Error(err))
-			return err
 		}
-	}
-
-	if h.hub != nil {
```

Comment:
This condition would always evaluate to true, causing a return at this point, because either the hub was set above or an error was returned.